galleon-plugins/src/main/java/org/wildfly/galleon/plugin/Utils.java

```diff
@@ -55,6 +55,10 @@
  */
 public class Utils {
 
+    private static final String EXPRESSION_PREFIX = "${";
+    private static final String EXPRESSION_SUFFIX = "}";
+    private static final String EXPRESSION_ENV_VAR = "env.";
+    private static final String EXPRESSION_DEFAULT_VALUE_SEPARATOR = ";";
```

Comment:
This introduces a custom syntax vs other expression resolution WildFly supports, where we use ':'

Is this due to concern about ':' showing up as part of an expression key?

Reply:
@bstansberry , the ':' character is present in the Maven coordinates as separator in the file entries. We have code that split the string on ':' and are not expecting to have expression. I think that it is safer to use a different separator.

Reply:
Could you resolve any expression(s) on the whole line before splitting the GAV?

Reply:
something like https://github.com/wildfly/wildfly-core/blob/73db5a61bacd36906f4026280b56a6c0e624f655/controller/src/main/java/org/jboss/as/controller/ExpressionResolverImpl.java#L175

So you would end up with:

```java
String[] parts = parseAndResolve(str).split(":");
```

Reply:
or maybe we should add a dependency to wildfly-common and use its expression resolver: https://github.com/wildfly/wildfly-common/blob/de4ee0986582b61c2b677dd56ba07238e6bce893/src/main/java/org/wildfly/common/expression/Expression.java

We would end up with something like

```java
Expression expr = Expression.compile(str, flags);
String resolvedStr = expr.evaluateWithPropertiesAndEnvironment(true)
String[] parts = resolvedStr.split(":"); 
```

Reply:
The common depenency is explicitly excluded from the galleon-plugins shaded jar. I would prefer not include this dependency. If we want the ':' separator, I will implement a parser in the galleon plugins to handle it.

Reply:
I am supporting the ":" for default value separator with a rework of artifact parsing.